lock/lock.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lock
+
+import (
+	"github.com/pingcap/parser/model"
+	"github.com/pingcap/parser/mysql"
+	"github.com/pingcap/tidb/infoschema"
+	"github.com/pingcap/tidb/infoschema/perfschema"
+	"github.com/pingcap/tidb/sessionctx"
+)
+
+// Checker uses to check tables lock.
+type Checker struct {
+	ctx sessionctx.Context
+	is  infoschema.InfoSchema
+}
+
+// NewChecker return new lock Checker.
+func NewChecker(ctx sessionctx.Context, is infoschema.InfoSchema) *Checker {
+	return &Checker{ctx: ctx, is: is}
+}
+
+// CheckTableLock uses to check table lock.
+func (c *Checker) CheckTableLock(db, table string, privilege mysql.PrivilegeType) error {
+	if db == "" || table == "" {
+		return nil
+	}
+	// Below database are not support table lock.
+	if db == infoschema.LowerName || db == perfschema.LowerName || db == mysql.SystemDB {
+		return nil
+	}
+	switch privilege {
```

Comment:
Do we need to consider about  "drop database"? Like the follows:
```
 create database db;
```
session1:
```
use db;
lock tables t write;
```
session2:
```
 drop database db;
```

Reply:
Great, Thanks.

Reply:
Do we need to consider the "truncate table" and other DDL  statements?

Reply:
I think `checkLockTpMeetPrivilege` already do this job.

Reply:
Got it.